### PR TITLE
Issue-450: Convert Pytest-CVP Hardware Temperature test case to Vane

### PR DESCRIPTION
**Pull request by Ashutosh-GSLAB**
# Please include a summary of the changes

test_definition.yaml: Updated test def with testcase NRFU6.11
test_definition.yaml.j2: Updated J2 with testcase NRFU6.11
master_def.yaml: Updated master def with testcase NRFU6.11
test_syslog_functionality.py: Added python file for testcase NRFU6.11

# Any specific logic/part of code you need extra attention on

validating system temperature sensors details

# Include the Issue number and link
#450 

# List/Attach any dependencies/past issues that are required for this change/provide context

# Type of change

- - [ ] Bug fix (non-breaking change which fixes an issue)
- - [ ] New feature (non-breaking change which adds functionality)
- - [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- - [ ] This change requires a documentation update
- - [X] Other (NRFU testcases)

# Effort required on reviewers end

- - [ ] Easy
- - [X] Medium
- - [ ] Hard 

# How Has This Been Tested?

Unit tests and report :

Pass Report:

1.  pass report ( pass without j2 )

[without_j2.docx](https://github.com/aristanetworks/vane/files/12607416/without_j2.docx)

[without_j2.zip](https://github.com/aristanetworks/vane/files/12607428/without_j2.zip)

2. pass report ( pass with j2 )

[with_j2.docx](https://github.com/aristanetworks/vane/files/12607421/with_j2.docx)

[with_j2.zip](https://github.com/aristanetworks/vane/files/12607430/with_j2.zip)

Fail Report:
Note: Tested with using hardcoded output

3. Fail report when hardware status are not ok and sensors are met.

[report_2309081218.docx](https://github.com/aristanetworks/vane/files/12556397/report_2309081218.docx)

[report.html_fail_1.zip](https://github.com/aristanetworks/vane/files/12556404/report.html_fail_1.zip)

4. assert fail when power supplies sensors detail are not found.

[report_2309051741.docx](https://github.com/aristanetworks/vane/files/12523292/report_2309051741.docx)

5.  assert fail when temperature sensors detail are not found.

[report_2309051744.docx](https://github.com/aristanetworks/vane/files/12523312/report_2309051744.docx)

Html reports :

[html_reports.zip](https://github.com/aristanetworks/vane/files/12523662/html_reports.zip)

## New feature

 
    
# CI pipeline result

- - [ ] Pass
- - [ ] Fail
  
  If it fails, explain the reason and whether or not we should ignore the failure
  
# Verify Documentation Update

  
    
# Additional comments

### CODE REVIEW

File: nrfu_tests/master_def.yaml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+
+common_data:
```
Comment (AbhijeetGSLAB) [inline nrfu_tests/master_def.yaml:2]: Update the master def as per placeholder MR

File: nrfu_tests/test_system_temperature_sensors.py [hunk 2]
```diff
@@ -0,0 +1,155 @@
+# Copyright (c) 2022 Arista Networks, Inc.  All rights reserved.
```
Comment (AbhijeetGSLAB) [inline nrfu_tests/test_system_temperature_sensors.py:1]:
```suggestion
# Copyright (c) 2023Arista Networks, Inc.  All rights reserved.

```

File: nrfu_tests/test_system_temperature_sensors.py [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+                    temp_status = current_temperature + 5 >= overheat_threshold
+
+                    tops.expected_output["power_supplies_sensors"].update(
+                        {name: {"hardware_status": "ok", "temp_is_getting_high": True}}
+                    )
+                    tops.actual_output["power_supplies_sensors"].update(
+                        {
+                            name: {
+                                "hardware_status": sensor["hwStatus"],
+                                "temp_is_getting_high": temp_status,
```
Comment (AbhijeetGSLAB) [inline nrfu_tests/test_system_temperature_sensors.py:99]: The key can be temperature_status

File: nrfu_tests/test_system_temperature_sensors.py [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+    """
+    Testcase for verification of system temperature sensors functionality.
+    """
+
+    dut_parameters = tests_tools.parametrize_duts(TEST_SUITE, test_defs, dut_objs)
+    test_duts = dut_parameters["test_validate_system_temperature_sensors"]["duts"]
+    test_ids = dut_parameters["test_validate_system_temperature_sensors"]["ids"]
+
+    @pytest.mark.parametrize("dut", test_duts, ids=test_ids)
+    def test_validate_system_temperature_sensors(self, dut, tests_definitions):
```
Comment (AbhijeetGSLAB) [inline nrfu_tests/test_system_temperature_sensors.py:29]:
```suggestion
    def test_system_temperature_sensors(self, dut, tests_definitions):

```

File: nrfu_tests/test_system_hardware_temperature.py [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+    def test_system_temperature_sensors(self, dut, tests_definitions):
+        """
+        TD: Testcase for verification of system temperature sensors functionality
+        Args:
+            dut(dict): details related to a particular DUT
+            tests_definitions(dict): test suite and test case parameters.
+        """
+        tops = tests_tools.TestOps(tests_definitions, TEST_SUITE, dut)
+        self.output = ""
+        tops.actual_output = {"temp_sensors": {}, "power_supplies_sensors": {}}
```
Comment (vishwajeetgslab) [inline nrfu_tests/test_system_hardware_temperature.py:38]:
```suggestion
        tops.actual_output = {"sensors": {"temperature_sensors": {}, "power_supply_sensors": {}}}

```

File: nrfu_tests/test_system_hardware_temperature.py [hunk 6]
```diff
@@ -0,0 +1,152 @@
...
+            TS: Running "show version" commands on DUT.
+            Verifying the 'vEOS' device is not present in output.
+            """
+            version_output = dut["output"]["show version"]["json"]
+            logger.info(
+                "On device %s: Output of 'show version' command is: \n%s\n",
+                tops.dut_name,
+                version_output,
+            )
+            self.output += f"\n\nOutput of 'show version' command is: \n{version_output}"
```
Comment (vishwajeetgslab) [inline nrfu_tests/test_system_hardware_temperature.py:55]:
```suggestion
            self.output += f"\nOutput of 'show version' command is: \n{version_output}"

```

File: nrfu_tests/test_system_hardware_temperature.py [hunk 7]
```diff
@@ -0,0 +1,152 @@
...
+                    current_temperature = sensor["currentTemperature"]
+                    overheat_threshold = sensor["overheatThreshold"]
+                    temp_status = current_temperature + 5 >= overheat_threshold
+
+                    tops.expected_output["temp_sensors"].update(
+                        {description: {"temp_is_getting_high": True}}
+                    )
+                    tops.actual_output["temp_sensors"].update(
+                        {description: {"temp_is_getting_high": temp_status}}
+                    )
```
Comment (vishwajeetgslab) [inline nrfu_tests/test_system_hardware_temperature.py:114]: Can this repeated loop be avoided.

File: nrfu_tests/test_system_hardware_temperature.py [hunk 8]
```diff
@@ -0,0 +1,152 @@
...
+            for sensor in other_temp_sensors:
+                # account for PhyAlaska chips that don't give current temp in 7020TR
+                description = sensor["description"]
+                if "PhyAlaska" not in description:
+                    current_temperature = sensor["currentTemperature"]
+                    overheat_threshold = sensor["overheatThreshold"]
+                    temp_status = current_temperature + 5 >= overheat_threshold
+
+                    tops.expected_output["temp_sensors"].update(
+                        {description: {"temp_is_getting_high": True}}
```
Comment (vishwajeetgslab) [inline nrfu_tests/test_system_hardware_temperature.py:110]:
```suggestion
                        {description: {"overheat_threshold_met": False}}

```

File: nrfu_tests/test_system_hardware_temperature.py [hunk 9]
```diff
@@ -0,0 +1,152 @@
...
+        tops.output_msg = "System temperature sensors detail are correct."
+
+        try:
+            """
+            TS: Running "show version" commands on DUT.
+            Verifying the 'vEOS' device is not present in output.
+            """
+            version_output = dut["output"]["show version"]["json"]
+            logger.info(
+                "On device %s: Output of 'show version' command is: \n%s\n",
```
Comment (vishwajeetgslab) [inline nrfu_tests/test_system_hardware_temperature.py:51]:
```suggestion
                "On device %s, Output of 'show version' command is: \n%s\n",

```

**Review by vishwajeetgslab — CHANGES_REQUESTED** [review]

File: nrfu_tests/test_system_hardware_temperature.py [hunk 10]
```diff
@@ -0,0 +1,249 @@
...
+                )
+                for sensor_detail, details_value in sensor_details.items():
+                    actual_sensor = actual_sensor_details.get(sensor_detail)
+
+                    temperature_detail = (
+                        current_threshold_temp_detail["power_supplies_sensors"]
+                        .get(sensor)
+                        .get(sensor_detail)
+                    )
+                    if details_value == actual_sensor:
```
Comment (vishwajeetgslab) [inline nrfu_tests/test_system_hardware_temperature.py:89]:
```suggestion
                    if expected_sensor_details == actual_sensor_details :

```

**Review by vishwajeetgslab — CHANGES_REQUESTED** [review]

**Review by AbhijeetGSLAB — APPROVED** [review]

File: nrfu_tests/test_system_hardware_temperature.py [hunk 11]
```diff
@@ -0,0 +1,260 @@
...
+                for sensor_detail, details_value in sensor_details.items():
+                    actual_sensor = actual_sensor_details.get(sensor_detail)
+                    temperature_detail = (
+                        current_threshold_temp_detail["power_supplies_sensors"]
+                        .get(sensor)
+                        .get(sensor_detail)
+                    )
+                    if details_value == actual_sensor:
+                        continue
+                    output_msg += f"For {sensor_detail}:"
```
Comment (geetanjalimanegslab) [inline nrfu_tests/test_system_hardware_temperature.py:96]:
```suggestion
                    output_msg += f"{sensor_detail}:"

```

File: nrfu_tests/test_system_hardware_temperature.py [hunk 12]
```diff
@@ -0,0 +1,260 @@
...
+                        output_msg += (
+                            " Expected hardware status should be"
+                            f" '{details_value['hardware_status']}' however in actual it is"
+                            f" found as '{actual_sensor['hardware_status']}'\n"
+                        )
+                        continue
+                    output_msg += (
+                        " Current sensor temperature is"
+                        f" '{temperature_detail['current_temperature']}(c)' which is greater"
+                        " than Threshold temperature"
```
Comment (geetanjalimanegslab) [inline nrfu_tests/test_system_hardware_temperature.py:107]:
```suggestion
                        " than threshold temperature"

```

File: nrfu_tests/test_system_hardware_temperature.py [hunk 13]
```diff
@@ -0,0 +1,260 @@
...
+            temp_sensor(bool): Flag to check the hardware status of the temperature sensor.
+            If it is set to True then the hardware status for a particular sensor will check
+        Return:
+            dict: Actual output of hardware status
+            dict: Current and overheat threshold temperature detail for power supply
+            and temperature sensors
+        """
+        actual_output = {}
+        temperature_details = {}
+        current_temperature = sensor_detail["currentTemperature"]
```
[the remark this answers is not in the document]
Comment (Ashutosh-GSLAB) [reply nrfu_tests/test_system_hardware_temperature.py:38]: we will get the current temperature always. so I thing we can access it without using get method.

**Review by geetanjalimanegslab — APPROVED** [review]

**Review by vishwajeetgslab — APPROVED** [review]

**Review by kgrozis-arista — APPROVED** [review]

**Pull request merged**